Add Deck class

daelynj opened the pull request:
Add a Deck class to generate an array of Card objects to simulate a
deck of playing cards and then shuffle.

Add a deal method to deal 5 cards simultaneously removing each dealt
card from the Deck object.

daelynj committed:
Add Deck class

Add a Deck class to generate an array of Card objects to simulate a
deck of playing cards and then shuffle.

Add a deal method to deal 5 cards simultaneously removing each dealt
card from the Deck object.

daelynj committed:
Merge pull request #1 from daelynj/add-card-class

Add Card class

swcraig commented on lib/poker_hands/deck.rb:
```diff
@@ -0,0 +1,27 @@
+
+class Deck
+  def initialize
+    @hand_size = 5
```
Is there a reason why a `Deck` should know about the size of a hand?

daelynj replied:
Would you advocate for the `deal` method to have a parameter to receive `hand_size`?

swcraig replied:
I would consider moving the idea of dealing cards out of this class entirely.

daelynj committed:
Remove call to to_s in deal method

The deal method no longer relies on the to_s method from Cards and
returns a hand array of Card objects instead.

daelynj committed:
Add parameter to deal

The deal method now receives a parameter of hand_size because a Deck
object doesn't need to know the default size of a hand of cards.
